topsort-api-v2.yml

```diff
@@ -629,6 +682,108 @@ components:
             format: double
           minItems: 1
           maxItems: 10000
+ 
+    SponsoredBrandAuctionRequest:
+      type: object
+      description: |
+        Describes the intent of running a sponsored brand auction.
+
+        Exactly **one** of the following fields must be set:
+          * `products`
+          * `category`
+      properties:
+        winners:
+          type: integer
+          format: int32
+          minimum: 1
+          description: Specifies the maximum number of auction winners that should be returned.
+        category:
+          $ref: '#/components/schemas/SingleCategory'
+        products:
+          $ref: '#/components/schemas/Products'
+      required:
+        - winners
+    
+    SponsoredBrandAuctionResult:
+      type: object
+      properties:
+        winners:
+          type: array
+          items:
+            $ref: '#/components/schemas/SponsoredBrandWinner'
+          description: >
+            Array of winner objects in order from highest to lowest bid.
+            It will be empty if there were no qualifying bids or if there was an error.
+        error:
+          type: boolean
+          description: A boolean indicating whether this auction was resolved successfully.
+          example: false
+      required:
+        - winners
+        - error
+
+    SponsoredBrandWinner:
+      type: object
+      required:
+        - rank
+        - productId
+        - resolvedBidId
+        - assets
+      properties:
+        rank:
+          type: integer
+          format: int32
+          description: >
+            Where the product's bid ranked in the auction.
+            One-based, so the product with rank 1 won the auction.
+            In an auction response, the winners array is sorted so rank will match the entry's index.
+          minimum: 1
+        productId:
+          type: string
+          description: |-
+            The marketplace's ID of the winning entity, depending on the target of the campaign.
+          example: p_Mfk15
+        resolvedBidId:
+          type: string
+          description: An opaque Topsort ID to be used when this item is interacted with.
+          example: WyJiX01mazE1IiwiMTJhNTU4MjgtOGVhZC00Mjk5LTgzMjctY2ViYjAwMmEwZmE4IiwibGlzdGluZ3MiLCJkZWZhdWx0IiwiIl0=
```

Comment:
For display purposes we could shorten this id.

And maybe document it, indicating that no assumptions can be made about the alphabet or the length.

Reply:
I don't think the example is displayed in readMe. I'll keep it like this because it's how we document it for v2/auctions. I think that 'opaque' conveys that no assumption can be made.